packages/context/src/binding.ts

```diff
@@ -503,9 +504,18 @@ export class Binding<T = BoundValue> {
       debug('Bind %s to class %s', this.key, ctor.name);
     }
     this._type = BindingType.CLASS;
-    this._setValueGetter((ctx, options) =>
-      instantiateClass(ctor, ctx, options.session),
-    );
+    this._setValueGetter((ctx, options) => {
+      const instOrPromise = instantiateClass(ctor, ctx, options.session);
+      if (!options.asProxyWithInterceptors) return instOrPromise;
+      return transformValueOrPromise(instOrPromise, inst => {
```

Comment:
Please extract this part into a named function and keep `this._setValueGetter` callback size down to 3-5 lines of code.